Added 'Shuffle Message Order' button to settings.

t7ewithp opened the pull request:
Added a shuffle message order button to settings, that when toggled will cause the program to select a random message from the full set as its next message, rather than the next one in order.

I didn't change the screenshot, but that should probably be changed eventually as well.

t7ewithp committed:
Added a simple .gitignore file

t7ewithp committed:
Added button for shuffle feature.

t7ewithp committed:
Added shuffle functionality.

t7ewithp committed:
shuffle should default to off, oops

Go1den commented:
streamTickerWindow.py: Line 111-113, can we eliminate the redundancy in the logic here?

Something like this instead:
if len(self.messages) > 0:
    if self.settings.messageSettings.shuffle:
        self.currentIndex = randrange(len(self.messages))
    else:
        self.currentIndex = (self.currentIndex + 1) % len(self.messages)
else:
    self.currentIndex = 0

As we discussed on Discord, let's make shuffle a window setting instead of a message setting, since message settings can be changed per message and shuffle applies to the window itself.

Also this is completely my fault and something I already plan to fix before I do a new release, but if the settings file is missing a shuffle setting (for example, someone uses their settings from a previous version), the app will break because of a KeyError on the missing shuffle parameter. The KeyError will occur in streamTickerWindow.py's getSettings method. I think I already have code to fix this though so don't worry about it for purposes of your PR.

t7ewithp committed:
Move shuffle to Window Settings

t7ewithp committed:
Improved next message logic.

t7ewithp commented:
OK, made the 2 requested changes. It would be nice to at least prompt for a version update before crashing on wrong settings file, but yeah that's not in this PR's scope I think.

t7ewithp committed:
Changed the shuffle system to show each message once before reshuffling

This was suggested by go1den and it was agreed by claude that this was
what he wanted so, it's what we do.
